src/ts/components/MatchOverlay.tsx

```diff
@@ -40,6 +41,8 @@ const matchOverlay = <TPluginState extends IPluginState>({
     null
   );
   const [arrowElement, setArrowElement] = useState<HTMLDivElement | null>(null);
+  const [showMatch, setShowMatch] = useState<boolean>(false);
+  const debounceShowMatch = useRef(debounce(setShowMatch, 200))
```

Comment:
Oh, this is really interesting. So `useRef` will only accept the first result of the `debounce` call, and use it for the entire lifecycle.

It's kindof weird to me that the result of subsequent calls to `debounce` get dropped on the floor, but this solution is extremely neat. Hooks, oh boy.